### ISSUE TITLE
Systemtray context menu empty in fedora mate

### ISSUE THREAD

**Issue by coozoo** [goldendict/goldendict#1155]
Hi
After  goldendict update context menu of systemtray icon is empty. Appears like a tall thin box.

Version without bug:
goldendict-1.5-0.16.RC2.fc29
![goldendict-1 5-0 16 RC2 fc29](https://user-images.githubusercontent.com/25594311/58068670-e7769880-7b9a-11e9-8778-c50d418848a1.png)

Version with bug:
goldendict-1.5-0.19.RC2.fc29
![goldendict-1 5-0 19 RC2 fc29](https://user-images.githubusercontent.com/25594311/58068692-feb58600-7b9a-11e9-9f8a-060f7118fe9b.png)

**Comment by vedgy** [comment]
This is a duplicate of #907 and #1097. You can find workarounds and possible fixes in descriptions and comments to these issues.

As for the regression claim, perhaps the older Fedora package depended on Qt4 and the newer one depends on Qt5? If yes, then it probably means that Qt5 notification area icon menus do not work out of the box on your system/desktop environment. You may have to apply some system tweaks. This menu works perfectly with the latest Goldendict and Qt5 versions on my Manjaro system with Xfce DE.

**Comment by vedgy** [comment]
As you can see [here](https://github.com/goldendict/goldendict/blame/13a321190fd32f7125aedded8843a8831717bb86/mainwindow.cc#L798), the relevant `trayIconMenu` code has not changed since 2009.

**Comment by coozoo** [comment]
Thank, sorry I've tried to find duplicates but as I see failed in that )

You are right about that it wasn't changed.

After small investigation, now I see the difference
it's qt version 
goldendict-1.5-0.16.RC2.fc29 - built with qt4
goldendict-1.5-0.19.RC2.fc29 - built with qt5

I've tried to built goldendict with qt  5.11.3-1.fc29 and got the same result.
But I have my application based on qt tray example where menu works fine and built with the same qt version...

so now I will try to figure out where the problem (not sure about my abilities :) )....

**Comment by vedgy** [comment]
There is not much `trayIcon`-related code and it's all in `mainwindow.cc`. I'm sure you will find the offending difference. Perhaps it's caused by the dummy tray icon created [here](https://github.com/goldendict/goldendict/blob/13a321190fd32f7125aedded8843a8831717bb86/mainwindow.cc#L588). Try moving that code down just before [this line](https://github.com/goldendict/goldendict/blob/13a321190fd32f7125aedded8843a8831717bb86/mainwindow.cc#L794) and removing the first of the `trayIcon->show();` calls. Also try removing the `updateTrayIcon();` call from `MainWindow::MainWindow()`.

**Comment by coozoo** [comment]
Hi, 
I've tried to moving and it works only if put "dummy" tray after makeDictionaries(); call. As for me it brokes idea to show user that we are doing something.
So I've made fix like above, some dirty trick :) removing dummy and creating new one the same,
